ratatui-core/src/symbols/merge.rs

```diff
@@ -300,10 +300,14 @@ impl MergeStrategy {
     /// [Box Drawing Unicode block]: https://en.wikipedia.org/wiki/Box_Drawing
     /// [`Cell::merge_symbol`]: crate::buffer::Cell::merge_symbol
     pub fn merge<'a>(self, prev: &'a str, next: &'a str) -> &'a str {
+        // Replace should always just return the last symbol.
+        if self == Self::Replace {
+            return next;
+        }
         let (Ok(prev_symbol), Ok(next_symbol)) =
             (BorderSymbol::from_str(prev), BorderSymbol::from_str(next))
         else {
-            return next;
+            return prev;
```

Comment:
I think this actually needs to work out which of the two items is a border and preferentially choose the non-border character.
- "A" + "-" => "A"
- "-" + "A" => "A"

We should maybe also consider whether we need to rename any of the variant names (or add extra ones) to make this clear (and allow for future behavior where this is not the case). (Not 100% sure of this right now though).